### PR TITLE
Fixed file_read_partitions so that it skips unused partition entries

### PR DESCRIPTION
**Pull request by mzhong1**
The file_read_partitions function creates partitions for empty partition array entries.  This commit skips over the empty partition entries

### CODE REVIEW

File: src/header.rs [hunk 1]
```diff
@@ -276,23 +276,24 @@ pub(crate) fn file_read_header(file: &mut File, offset: u64) -> Result<Header> {
...
-        part_start: u64::from_le_bytes( read_exact_buff!(pstart, reader, 8) ),
-        num_parts: u32::from_le_bytes( read_exact_buff!(nparts, reader, 4) ),
-        part_size: u32::from_le_bytes( read_exact_buff!(partsize, reader, 4) ),
-        crc32_parts: u32::from_le_bytes( read_exact_buff!(crc32parts, reader, 4) ),
+        part_start: u64::from_ne_bytes(read_exact_buff!(pstart, reader, 8)),
+        // Note: this will always return the total number of partition entries
+        // in the array, not how many are actually used
+        num_parts: u32::from_ne_bytes(read_exact_buff!(nparts, reader, 4)),
+        part_size: u32::from_ne_bytes(read_exact_buff!(partsize, reader, 4)),
+        crc32_parts: u32::from_ne_bytes(read_exact_buff!(crc32parts, reader, 4)),
```
Comment (lucab) [inline src/header.rs]: All these endianness changes seems unrelated to the topic of this PR. Why are you changing it?

File: src/macros.rs [hunk 2]
```diff
@@ -1,18 +1,15 @@
-
```
Comment (lucab) [inline src/macros.rs]: If all these changes are the results of latest rustfmt, it would be useful to just run `cargo fmt` on current master branch and just submit that as its own PR.

File: src/partition.rs [hunk 3]
```diff
@@ -227,31 +227,41 @@ pub(crate) fn file_read_partitions(
     let mut parts: BTreeMap<u32, Partition> = BTreeMap::new();
 
     trace!("scanning {} partitions", header.num_parts);
+    debug!("Number of parts {:?}\n\n", header.num_parts);
```
Comment (lucab) [inline src/partition.rs]: The `trace!` line above already cover it, please drop this and enable the relevant log level in the application.

File: src/header.rs (hunk 1 above)
Comment (Quyzi) [reply src/header.rs]: GPT spec is little endian. I don't understand why these changes.

Comment (mzhong1) [reply src/header.rs]: I can remove those -> I was initially concerned about the endianness (I thought that was why the partition thing was having issues, I can remove those if you'd like?)

File: src/partition.rs (hunk 3 above)
Comment (mzhong1) [reply src/partition.rs]: Will do!

File: src/header.rs (hunk 1 above)
Comment (mzhong1) [reply src/header.rs]: Alright, I returned those from_ne_bytes back to from_le_bytes

File: src/macros.rs (hunk 2 above)
Comment (mzhong1) [reply src/macros.rs]: Yeah, that was unintentional, whenever I saved it would automatically reformat it, and I might have done a git add . I reverted it

**Comment by Quyzi** [comment]
Looks good to me! Thanks for the fixes.
